components/Upload.tsx

```diff
@@ -1,31 +1,97 @@
 import React, { useState } from "react";
-import { View, Text, StyleSheet, Dimensions, Image } from "react-native";
+import {
+  View,
+  Text,
+  StyleSheet,
+  Dimensions,
+  Image,
+  ToastAndroid,
+} from "react-native";
 import { Pressable } from "react-native";
 import { TouchableOpacity } from "react-native-gesture-handler";
-import {ArrowUp} from "../assets/svgs/svg"
+import {ArrowUp} from "../assets/svgs/svg";
 import * as ImagePicker from "expo-image-picker";
-export default function Upload({ document, }) {
-    const [image, setImage] = useState(null);
+import axios from "axios";
+import Constants from "expo-constants";
+import { useContext } from "react";
+import { AuthContext } from "../context/AuthContext";
+export default function Upload(props) {
+  let url = Constants?.manifest?.extra?.URL;
+  axios.defaults.baseURL = url;
+  const { authData } = useContext(AuthContext);
+  const [showLoader, setShowLoader] = useState(false);
+  const [image, setImage] = useState(null);
 
-     const pickImage = async () => {
-       // No permissions request is necessary for launching the image library
-       let result = await ImagePicker.launchImageLibraryAsync({
-         mediaTypes: ImagePicker.MediaTypeOptions.All,
-         allowsEditing: true,
-         aspect: [4, 3],
-         quality: 1,
-       });
+  const pickImage = async () => {
+    let result = await ImagePicker.launchImageLibraryAsync({
+      mediaTypes: ImagePicker.MediaTypeOptions.All,
+      allowsEditing: true,
+      aspect: [4, 3],
+      quality: 1,
+    });
+    if (!result.cancelled) {
+      setShowLoader(true)
+    // ImagePicker saves the taken photo to disk and returns a local URI to it
+    let localUri = result.uri;
+    let filename = localUri.split("/").pop();
+    // Infer the type of the image
+    let match = /\.(\w+)$/.exec(filename);
+    let type = match ? `image/${match[1]}` : `image`;
 
-       if (!result.cancelled) {
-         setImage(result.uri)
-       }
-     };
+    // Upload the image using the fetch and FormData APIs
+    let formData = new FormData();
+    formData.append("document", { uri: localUri, type:type, name: filename });
+    formData.append("type", props.type)
+      let res = await fetch(
+        "https://altara-customer-play-api.herokuapp.com/api/v1/document/upload",
```

Comment:
can we put the base api in an ENV, so we can swap when building for production?

Reply:
done